packages/loopring_v3/contracts/impl/libexchange/ExchangeBlocks.sol

```diff
@@ -367,130 +218,98 @@ library ExchangeBlocks
+            // uint24 operatorAccountID = data.toUint24(offset);
+            offset += 3;
 
-                // Update the onchain state
-                consumeConditionalTransfer(
-                    S,
-                    operatorAccountID,
-                    transferData
-                );
+            // Run over all conditional transactions
+            uint previousTransactionOffset = 0;
+            for (uint i = 0; i < txAuxiliaryData.length; i++) {
+                // Each conditional transaction needs to be processed from left to right
+                uint transactionOffset = offset + txAuxiliaryData[i].txIndex * ExchangeData.TX_DATA_AVAILABILITY_SIZE();
+                require(transactionOffset > previousTransactionOffset, "AUXILIARYDATA_INVALID_ORDER");
+
+                // Get the transaction data
+                bytes memory txData = data.slice(transactionOffset, ExchangeData.TX_DATA_AVAILABILITY_SIZE());
+
+                // Process the transaction
+                uint txFeeETH = 0;
+                ExchangeData.TransactionType txType = ExchangeData.TransactionType(txData.toUint8(0));
+                if (txType == ExchangeData.TransactionType.TRANSFER) {
+                    txFeeETH = TransferTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else if (txType == ExchangeData.TransactionType.WITHDRAWAL) {
+                    txFeeETH = WithdrawTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else if (txType == ExchangeData.TransactionType.DEPOSIT) {
+                    txFeeETH = DepositTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else if (txType == ExchangeData.TransactionType.ACCOUNT_NEW) {
+                    txFeeETH = NewAccountTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else if (txType == ExchangeData.TransactionType.ACCOUNT_UPDATE) {
+                    txFeeETH = AccountUpdateTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else if (txType == ExchangeData.TransactionType.ACCOUNT_TRANSFER) {
+                    txFeeETH = OwnerChangeTransaction.process(
+                        S,
+                        ctx,
+                        txData,
+                        txAuxiliaryData[i].data
+                    );
+                } else {
```

Comment:
Only conditional transactions need onchain logic, which we don't have for NOOP because those transactions don't do anything, or trades because they always need EdDSA signatures. We only run over all conditional transactions onchain, all other transactions are skipped over.